src/styles/styles.js

```diff
@@ -1328,8 +1328,9 @@ const styles = {
         flexShrink: 0,
         color: themeColors.textSupporting,
         fontSize: variables.fontSizeSmall,
-        height: 24,
+        height: 20,
         lineHeight: variables.lineHeightXLarge,
+        paddingBottom: 4,
```

Comment:
I think we want to get rid of the padding bottom. However, for image attachments, if the image attachment is the first thing in the message group (it comes right after the sender name) we should give it some top margin of say 4-8px.

Reply:
I just read the Slack thread linked to the issue and it seems like we also want to do some cleanup on the negative margins used for the message text, as well as add some padding for images. I'll add that to this PR.

That being said, I'm having issues aligning the sender name & date at the baseline. `alignItems: 'baseline'` is [supported in RN](https://reactnative.dev/docs/flexbox#align-items) and works well on browsers, but the `Pressable` component wrapping the sender name text is throwing off the alignment on iOS/android native since it's baseline is different from the text baseline. Here's how they behave:

**iOS**
![ios](https://user-images.githubusercontent.com/22219519/212763531-b3db0f65-7dbc-48e4-b7b6-15ef3513a714.png)

**android**
![android](https://user-images.githubusercontent.com/22219519/212763558-34e1ade3-f039-4bbb-80dd-9e1bb914a6dc.png)

I've spent some time trying to work around this issue, but no luck so far. Any suggestions?

Reply:
So after seeing this, it does feel like the sender is too close to the message, but maybe that's because the negative margin is still there. So let me know once you start playing around with that stuff too.

Reply:
>There was a proposal that seemed to fix the alignment by always making sure the big text (sender) and small text (timestamp) had the same lineheight and height. So maybe we could try that first?

I tried this and the other solutions in that issue but none of them seem to work. Maybe I'm holding something wrong?

I'd appreciate any other suggestions as I'm struggling to find a solution that doesn't add any padding/margins to the text.

Reply:
Hmm I am a bit stumped personally, maybe @aneequeahmad has some ideas given that he was able to solve this in his proposal in the linked issue above?

Reply:
@aneequeahmad would you mind taking a look at this one? I tried a few combinations of using the same `height` and `lineHeight` for the sender and/or date, but none of them seemed to work.

Reply:
@shawnborton Thank-you for mentioning me. This issue is fixed when i `remove the height from the div` containing `text'. 

https://github.com/Expensify/App/blob/4a3ed076bdf78a5f98b38aa8ecb195178e22d707/src/styles/styles.js#L1327-L1331

There are actually 3 scenarios when there are 2 text in a line and there is `text vertical center alignment issue`

**Case 1:** There is only one Text under View Ios: Use flex with alignItems/justifyContent to be more centered, Adr: height = lineHeight, includeFontPadding: false more centered

**Case 2:** There are two Texts in a line view and the two Text fonts are different, and the Text with a smaller font size does not need to use the height.

- The larger one is aligned using the above method. The smaller one can only be used with font-size, but not with the line-height and height attributes. In this case, both texts can be centered.

Case 3: There are two Texts in a line view and the two Text fonts are different, and the Text with a smaller font size must use the height.

- The one with the larger font size is aligned using the above method, and the smaller font size uses height with padding and includeFontPadding to achieve alignment.


Here is the video attached after the fix is applied.

**MacOs(Chrome)**


https://user-images.githubusercontent.com/36359331/213567065-392706c0-300e-430d-9d9a-0eb7a221be1a.mov


**Android**
![Screen Shot 2023-01-20 at 2 30 14 AM](https://user-images.githubusercontent.com/36359331/213567177-3b53041f-1a56-4112-88e4-cd203cc523d6.png)

Reply:
@aneequeahmad thanks for the comment! Sorry, but I'm still confused. Here's what I _think_ you are trying to say for Case 2:

- `chatItemMessageHeaderSender` (the large text) should have `height` = `lineHeight`
- `chatItemMessageHeaderTimestamp` (the small text) should have only `fontSize`

Is that right? If so, I tried it in this branch but the text is still not aligned

![Screenshot 2023-01-19 at 4 05 45 PM](https://user-images.githubusercontent.com/22219519/213582049-6e1b1762-7e85-412d-8a11-c23a076cd4a5.png)

Reply:
@luacmartins Actually  [Text with a smaller font size does not need to use the height](https://github.com/Expensify/App/pull/14304#:~:text=Text%20with%20a%20smaller%20font%20size%20does%20not%20need%20to%20use%20the%20height.) 

What you said above isn't what case 2 says[ (the large text) should have height = lineHeight](https://github.com/Expensify/App/pull/14304#:~:text=chatItemMessageHeaderSender%20(the%20large%20text)%20should%20have%20height%20%3D%20lineHeight).

Reply:
Sorry, I'm having trouble understanding those instructions. Would you mind posting a diff of the changes you used [here](https://github.com/Expensify/App/pull/14304#discussion_r1081890978)?

Reply:
```diff
 chatItemMessageHeaderTimestamp: { 
     flexShrink: 0, 
     color: themeColors.textSupporting, 
     fontSize: variables.fontSizeSmall, 
     height: 24, 
-        height: 24,
+        //height: 24,
         lineHeight: variables.lineHeightXLarge,
     },
 ```

Reply:
@luacmartins Did you understand what i said above ? Just need to remove the height from `chatItemMessageHeaderTimestamp`.

I saw in your last commit you've removed height and lineHeight as well from `chatItemMessageHeaderTimestamp` Which isn't aligning and moves the date 1-2px below the `name `as `lineHeight` was removed

Reply:
@aneequeahmad I pushed the changes adding `lineHeight` back to `chatItemMessageHeaderTimestamp`, but the text is still not aligned. Am I missing something else?

Reply:
BTW I'm removing the `paddingBottom` from `chatItemMessageHeaderSender` because of [this comment](https://github.com/Expensify/App/pull/14304#discussion_r1070665710)

Reply:
@luacmartins i just noticed you added `height: variable.lineHeightXLarge` and removed `paddingBottom: 4` i reverted this change and the fix is working.

### Screenshot:

![Screen Shot 2023-01-21 at 1 47 15 AM](https://user-images.githubusercontent.com/36359331/213802828-3e968fed-ad5b-4086-b25d-d457b9ffa465.png)

Reply:
Yea, I linked @shawnborton's comment above about removing `paddingBottom`. Any ides on how to align the items without the padding?

Reply:
I understand why you removed `paddingBottom` but i would say that i have learned from working on startup products pixel perfect apps that it isn't a good practice to give `height` to every element other than containers. Height should be automatically set as per design if we assign the right values of `fontSize, padding, margin etc.`

Setting `fixed height` causes these kind of issues on different devices or different platforms. If relative units are used layout is set perfectly as the design. 

For now i think it's fine to go with paddingBottom and remove the added height.

```diff
         fontSize: variables.fontSizeNormal,
         fontWeight: fontWeightBold,
         lineHeight: variables.lineHeightXLarge,
-        height: variables.lineHeightXLarge,
+        paddingBottom: 4,
         ...wordBreak.breakWord,
     },


```

Reply:
We can maybe have a discussion in open source to decide about these kind of Layout issues and define good practices and get the` input of other great engineers and designers` so that we have `pixel perfect UI` in the long run and we don't have to do back and forth on these kinda issues which mostly waste precious time and resources IMO. 

**Suggestion:**

Also, as `ExpensiConX is in March 2023` it would be great if we can have a session on design, the best practices and industry standards/conventions. 

cc: @shawnborton

Reply:
@shawnborton any guidance here? Should we keep `paddingBottom`?

Reply:
> Any ides on how to align the items without the padding?

I can look more into it but i'm afraid i don't completely understand why we need to remove the `paddingBottom` ?  If we decide to remove the paddingBottom then i think it would be difficult to use just only `lineHeight, Height, and fontsSize` to built the layout.

Reply:
Can you show me a side-by-side comparison of what it looks like with the paddingBottom, as well as without it?

Reply:
@shawnborton here is the side-by-side comparison of `without paddingBottom` and `with paddingBottom`

| Without paddingBottom  | With paddingBottom |
| ------------- | ------------- |
| ![Screen Shot 2023-01-23 at 3 45 22 AM](https://user-images.githubusercontent.com/36359331/213944415-325ced55-48bc-4220-a680-bcd30b232755.png) | ![Screen Shot 2023-01-23 at 3 46 35 AM](https://user-images.githubusercontent.com/36359331/213944423-0db4a9d7-737b-4da5-beb9-a24bb742d31d.png)

Reply:
Interesting. I'm not really sure what to make of this, why does paddingBottom make it aligned? 

Ideally this is what I am hoping to achieve, as it seems relatively simple. For messages, the avatar is 40px tall. So then ideally what we would do for the simple message case is have the sender + timestamp be 20px tall, and then the message below be 20px tall too (assuming it is one line). That gives us an alignment like so:
<img width="1106" alt="image" src="https://user-images.githubusercontent.com/2319350/213989950-c2c07a38-899d-4865-b96a-e0503c8a394a.png">

Reply:
So perhaps one thing we could play with is giving the sender + timestamp a height (or lineHeight) of 16px plus vertical padding of 2px, so that we get 2px + 16px + 2px = 20px tall. Then the message below should have a lineHeight of 20px.

Reply:
@shawnborton Looking into it why paddingBottom make it aligned. Also, this can be done what you said [above](https://github.com/Expensify/App/pull/14304#:~:text=Ideally%20this%20is,2px%20%3D%2020px%20tall). @luacmartins i just saw you pushed some commits in this branch. 
Should i look into it ? or you have investigated what @shawnborton suggested ?

Reply:
@shawnborton I updated the branch with the following:

1. Added `lineHeight: 20px` to the senders name
2. Added `lineHeight: 16px` and `paddingTop: 4px` to the timestamp

Now the header + single line message sums up to 40px and both seem to be aligned at the baseline:
![Screenshot 2023-01-23 at 4 19 09 PM](https://user-images.githubusercontent.com/22219519/214174974-23f4ce9c-a627-4ede-a945-27a327845aa8.png)

![Screenshot 2023-01-23 at 4 19 02 PM](https://user-images.githubusercontent.com/22219519/214174990-9b560d02-951e-4ea0-96b3-27942a7892b5.png)


In addition, I removed the negative margin from `chatItemMessage` and added a conditional 8px top margin to attachments when displayed with the header:

![Screenshot 2023-01-23 at 4 27 41 PM](https://user-images.githubusercontent.com/22219519/214175236-78b9e0b6-1e2f-4e1e-abbe-ee93c1f3fad8.png)

Let me know what you think!

Reply:
@aneequeahmad let's see what @shawnborton says about my changes above!

Reply:
That feels pretty great to me! Thanks for pushing through this one and getting to a good solution!

Reply:
@luacmartins Code changes looks good to me too. That's a pretty neat solution. @shawnborton Thank-you for your suggestion.

Reply:
Thank you both for the help!